src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -3887,6 +3890,8 @@ void ApplicationManagerImpl::CloseNaviApp() {
         commands::Command::SOURCE_SDL);
     UnregisterApplication(app_id, ABORTED);
   }
+
+  navi_app_to_stop_.pop_front();
```

Comment:
Seems this is more likely to cause a race condition, perhaps it should be lock-protected?